### ISSUE TITLE
Different behaviour on second build if the same easyconfig file is passed twice to the eb command line

### ISSUE THREAD

**Issue by gppezzi** [hpcugent/easybuild-framework#1636]
I pasted twice on the command line (by mistake) the list of ebs that I wanted to build  and I've noticed that the second build of the same package was failing (while the first one passed).

I'm giving here the logs of a failing build and a working one for debugging, but my feeling is that this could be fixed by just filtering the list to remove duplicates (or do you have an use case where it make sense to build stuff twice on the same command line?).

* Failed build: https://gist.github.com/gppezzi/eee4c9799dda57b9c87c
* Build OK: https://gist.github.com/gppezzi/ed7f602b71cae48f13fa

Here's the command line I've used to trigger this behaviour.
```
eb HPL-2.1-CrayCCE-2015.06-XC.eb HPL-2.1-CrayCCE-2015.11-XC.eb HPL-2.1-CrayGNU-2015.06-XC.eb HPL-2.1-CrayGNU-2015.11-XC.eb HPL-2.1-CrayIntel-2015.06-XC.eb HPL-2.1-CrayIntel-2015.11-XC.eb HPL-2.1-CrayCCE-2015.06-XC.eb HPL-2.1-CrayCCE-2015.11-XC.eb HPL-2.1-CrayGNU-2015.06-XC.eb HPL-2.1-CrayGNU-2015.11-XC.eb HPL-2.1-CrayIntel-2015.06-XC.eb
```

**Comment by boegel** [comment]
We could simply filter the provided list of easyconfigs for unique values, but the fact that the 2nd build failed worries me.

We do quite a bit of effort to try and ensure multiple builds done in the same session are isolated from each other. This suggests we missed something...

**Comment by boegel** [comment]
Filtered log shows that `$CFLAGS` is incorrectly defined in the 2nd round...

For some reason, the `-ftz` & co flags stick around.
This is a bug in the toolchain mechanism somewhere... :(

```
EB_HPL INFO building and installing HPL/2.1-CrayCCE-2015.06-XC...
environment INFO Environment variable CFLAGS set to -craype-verbose -O2 (previously undefined)
environment INFO Environment variable CFLAGS set to -craype-verbose -O2 (previously undefined)
environment INFO Environment variable CFLAGS set to -craype-verbose -O2 (previously undefined)
easyblock INFO COMPLETED: Installation ended successfully
EB_HPL INFO building and installing HPL/2.1-CrayCCE-2015.11-XC...
environment INFO Environment variable CFLAGS set to -craype-verbose -O2 (previously undefined)
environment INFO Environment variable CFLAGS set to -craype-verbose -O2 (previously undefined)
environment INFO Environment variable CFLAGS set to -craype-verbose -O2 (previously undefined)
easyblock INFO COMPLETED: Installation ended successfully
EB_HPL INFO building and installing HPL/2.1-CrayGNU-2015.06-XC...
environment INFO Environment variable CFLAGS set to -craype-verbose -O2 (previously undefined)
environment INFO Environment variable CFLAGS set to -craype-verbose -O2 (previously undefined)
environment INFO Environment variable CFLAGS set to -craype-verbose -O2 (previously undefined)
easyblock INFO COMPLETED: Installation ended successfully
EB_HPL INFO building and installing HPL/2.1-CrayGNU-2015.11-XC...
environment INFO Environment variable CFLAGS set to -craype-verbose -O2 (previously undefined)
environment INFO Environment variable CFLAGS set to -craype-verbose -O2 (previously undefined)
environment INFO Environment variable CFLAGS set to -craype-verbose -O2 (previously undefined)
easyblock INFO COMPLETED: Installation ended successfully
EB_HPL INFO building and installing HPL/2.1-CrayIntel-2015.06-XC...
environment INFO Environment variable CFLAGS set to -craype-verbose -O2 -ftz -fp-speculation=safe -fp-model source (previously undefined)
environment INFO Environment variable CFLAGS set to -craype-verbose -O2 -ftz -fp-speculation=safe -fp-model source (previously undefined)
environment INFO Environment variable CFLAGS set to -craype-verbose -O2 -ftz -fp-speculation=safe -fp-model source (previously undefined)
easyblock INFO COMPLETED: Installation ended successfully
EB_HPL INFO building and installing HPL/2.1-CrayIntel-2015.11-XC...
environment INFO Environment variable CFLAGS set to -craype-verbose -O2 -ftz -fp-speculation=safe -fp-model source (previously undefined)
environment INFO Environment variable CFLAGS set to -craype-verbose -O2 -ftz -fp-speculation=safe -fp-model source (previously undefined)
environment INFO Environment variable CFLAGS set to -craype-verbose -O2 -ftz -fp-speculation=safe -fp-model source (previously undefined)
easyblock INFO COMPLETED: Installation ended successfully
EB_HPL INFO building and installing HPL/2.1-CrayCCE-2015.06-XC...
environment INFO Environment variable CFLAGS set to -craype-verbose -O2 -ftz -fp-speculation=safe -fp-model source (previously undefined)
```

**Comment by boegel** [comment]
I can't reproduce this problem with other toolchains; I think it's specific to the Cray toolchains...

```
EB_HPL INFO building and installing HPL/2.1-goolf-1.7.20...
== building and installing HPL/2.1-goolf-1.7.20...
environment INFO Environment variable CFLAGS set to -O2 -march=native (previously undefined)
environment INFO Environment variable CFLAGS set to -O2 -march=native (previously undefined)
environment INFO Environment variable CFLAGS set to -O2 -march=native (previously undefined)
easyblock INFO COMPLETED: Installation ended successfully
== COMPLETED: Installation ended successfully
EB_HPL INFO building and installing HPL/2.1-intel-2015b...
== building and installing HPL/2.1-intel-2015b...
environment INFO Environment variable CFLAGS set to -O2 -xHost -ftz -fp-speculation=safe -fp-model source (previously undefined)
environment INFO Environment variable CFLAGS set to -O2 -xHost -ftz -fp-speculation=safe -fp-model source (previously undefined)
environment INFO Environment variable CFLAGS set to -O2 -xHost -ftz -fp-speculation=safe -fp-model source (previously undefined)
easyblock INFO COMPLETED: Installation ended successfully
== COMPLETED: Installation ended successfully
EB_HPL INFO building and installing HPL/2.1-iomkl-2015.02...
== building and installing HPL/2.1-iomkl-2015.02...
environment INFO Environment variable CFLAGS set to -O2 -xHost -ftz -fp-speculation=safe -fp-model source (previously undefined)
environment INFO Environment variable CFLAGS set to -O2 -xHost -ftz -fp-speculation=safe -fp-model source (previously undefined)
environment INFO Environment variable CFLAGS set to -O2 -xHost -ftz -fp-speculation=safe -fp-model source (previously undefined)
easyblock INFO COMPLETED: Installation ended successfully
== COMPLETED: Installation ended successfully
EB_HPL INFO building and installing HPL/2.1-goolf-1.7.20...
== building and installing HPL/2.1-goolf-1.7.20...
environment INFO Environment variable CFLAGS set to -O2 -march=native (previously undefined)
environment INFO Environment variable CFLAGS set to -O2 -march=native (previously undefined)
environment INFO Environment variable CFLAGS set to -O2 -march=native (previously undefined)
easyblock INFO COMPLETED: Installation ended successfully
== COMPLETED: Installation ended successfully
EB_HPL INFO building and installing HPL/2.1-intel-2015b...
== building and installing HPL/2.1-intel-2015b...
environment INFO Environment variable CFLAGS set to -O2 -xHost -ftz -fp-speculation=safe -fp-model source (previously undefined)
environment INFO Environment variable CFLAGS set to -O2 -xHost -ftz -fp-speculation=safe -fp-model source (previously undefined)
environment INFO Environment variable CFLAGS set to -O2 -xHost -ftz -fp-speculation=safe -fp-model source (previously undefined)
easyblock INFO COMPLETED: Installation ended successfully
== COMPLETED: Installation ended successfully
EB_HPL INFO building and installing HPL/2.1-iomkl-2015.02...
== building and installing HPL/2.1-iomkl-2015.02...
environment INFO Environment variable CFLAGS set to -O2 -xHost -ftz -fp-speculation=safe -fp-model source (previously undefined)
environment INFO Environment variable CFLAGS set to -O2 -xHost -ftz -fp-speculation=safe -fp-model source (previously undefined)
environment INFO Environment variable CFLAGS set to -O2 -xHost -ftz -fp-speculation=safe -fp-model source (previously undefined)
easyblock INFO COMPLETED: Installation ended successfully
== COMPLETED: Installation ended successfully
```

**Comment by gppezzi** [comment]
I had a similar problem when using --try-toolchain-version with this HPL + CrayCCE: it worked for HPL with CrayIntel and CrayGNU (with the same --try-tc command) and it failed for CCE.

Unfortunately I cannot reproduce this today, but I remember that building the eb file generated by try-toolchain (that failed) worked without any problems...

**Comment by boegel** [comment]
@gppezzi: were you using `--try-toolchain-version` and building HPL on top of CrayCCE and other Cray* toolchains in a single session?

**Comment by gppezzi** [comment]
I' ve managed to reproduce the try-toolchain issue and apparently it is unrelated to this problem, the CCE easyconfig files are actually different from the others (GNU/Intel). I did:

```eb HPL-2.1-CrayGNU-2015.06-XC.eb --try-toolchain=CrayCCE,2015.11-XC ```

The only hint I see here is that the toolchainopts from HPL CCE are missing.

**Comment by boegel** [comment]
OK, that does explain it, yes. It's called 'try' for a reason, it's only there for convenience.

**Comment by boegel** [comment]
fix in #1641, the problem was specific to the `Cray*` toolchains...
